docs/examples/tutorial/string/c_func.py

```diff
@@ -0,0 +1,30 @@
+from cython.cimports.libc.stdlib import malloc
+from cython.cimports.libc.string import strcpy, strlen
+
+hello_world = cython.declare(cython.p_char, 'hello world')
+n = cython.declare(Py_ssize_t, strlen(hello_world))
+
+
+# @cython.cfunc  # https://github.com/cython/cython/issues/4388
+def c_call_returning_a_c_string() -> cython.p_char:
```

Comment:
pretty sure that tests will fail on that if we will uncomment this (look #4388)

Reply:
IMHO, users should only have one place where types are defined. Either directly in the source file or in a .pxd file. Mixing both is just asking for annoyance.

We can add a code comment that makes it clear that the function is actually defined as a `cdef` function in the corresponding .pxd file – in case that's a) relevant enough and b) not clear from the surrounding text. Given that the topic is strings, we might fail on a).